lib/model.js

```diff
@@ -135,12 +135,21 @@ const _create = (data, definition) => {
     obj = EJSON.serialize(obj);
     const newEtag = etag(JSON.stringify(obj));
     const collection = definition.collection;
-    let newOrUpdatedModel;
+    let newOrUpdatedModel, result;
     if (!context.isNew()) {
       const query = { _id: obj._id, _etag: obj._etag };
       obj._etag = newEtag;
       debug(`updating ${collection}: ${JSON.stringify(obj)}`);
-      const result = await db.replace(collection, query, obj);
+      try {
+        result = await db.replace(collection, query, obj);
+      } catch (err) {
+        if (err.message.startsWith('E11000 duplicate key') && context.transformDuplicateKeyError) {
+          // this should throw an error.
+          context.transformDuplicateKeyError(obj, err);
```

Comment:
2 thoughts
1) I think we should just throw the result here (have the function return the new error)
2) you need to call it like `context.transformDuplicateKeyError.apply(obj, [err]);` (like beforeSave) - that way the `this` in the function is in fact the model object